src/common/scaled-font-buffer.c

```diff
@@ -42,9 +40,33 @@ _destroy(struct scaled_scene_buffer *scaled_buffer)
 	free(self);
 }
 
-static const struct scaled_scene_buffer_impl impl = {
+static bool str_equal(const char *a, const char *b)
+{
+	return a == b || (a && b && !strcmp(a, b));
+}
+
+static bool
+_equal(struct scaled_scene_buffer *scaled_buffer_a,
+	struct scaled_scene_buffer *scaled_buffer_b)
+{
+	struct scaled_font_buffer *a = scaled_buffer_a->data;
+	struct scaled_font_buffer *b = scaled_buffer_b->data;
+
+	return str_equal(a->text, b->text)
+		&& a->max_width == b->max_width
+		&& str_equal(a->font.name, b->font.name)
+		&& a->font.size == b->font.size
+		&& a->font.slant == b->font.slant
+		&& a->font.weight == b->font.weight
+		&& !memcmp(a->color, b->color, sizeof(float[4]))
+		&& !memcmp(a->bg_color, b->bg_color, sizeof(float[4]))
```

Comment:
`sizeof(a->color)` (or `b->color`) feels safer to me here in case we ever end up changing `float[4]` to something else.

Reply:
Done

Reply:
2nd one should be `sizeof(a->bg_color)` (its `sizeof(a->color)` currently).

Reply:
Oops. Fixed.